packages/types/api.d.ts

```diff
@@ -2119,6 +2127,147 @@ export interface IParserErrorMessageProvider {
+   *
+   * @param options.rules All parser rules of the grammar.
+   */
+  initialize?(options: { rules: Rule[] }): void
+
+  /**
+   * Builds a lookahead function for alternations/`OR` parser methods.
+   * The resulting function is able to compute which of the alternatives to choose while parsing.
+   *
+   * @param options.prodOccurrence The occurrence number of this `OR` within its rule.
+   *
+   * @param options.rule The rule that contains this `OR`.
+   *
+   * @param options.maxLookahead The maximum amount of lookahead for this `OR`.
+   *
+   * @param options.hasPredicates Whether any of the alternatives contain a predicate.
+   *
+   * @param options.dynamicTokensEnabled Whether dynamic tokens are enabled for this parser.
+   */
+  buildLookaheadForAlternation(options: {
+    prodOccurrence: number
+    rule: Rule
+    maxLookahead: number
+    hasPredicates: boolean
+    dynamicTokensEnabled: boolean
+  }): (orAlts?: IOrAlt<any>[] | undefined) => number | undefined
+
+  /**
+   * Builds a lookahead function for optional productions.
+   * The resulting function is able to compute whether to parse the production
+   * or to continue with the rest of the parser rule.
+   *
+   * @param options.prodOccurrence The occurrence number of this production within its rule.
+   *
+   * @param options.prodType The type of this production.
+   *
+   * @param options.rule The rule that contains this production.
+   *
+   * @param options.maxLookahead The maximum amount of lookahead for this production.
+   *
+   * @param options.dynamicTokensEnabled Whether dynamic tokens are enabled for this parser.
+   */
+  buildLookaheadForOptional(options: {
+    prodOccurrence: number
+    prodType: OptionalProductionType
+    rule: Rule
+    maxLookahead: number
+    dynamicTokensEnabled: boolean
+  }): () => boolean
+}
+
+export interface ILookaheadValidationError {
+  message: string
+  ruleName?: string
+}
+
+export type LookaheadSequence = TokenType[][]
+
+export type OptionalProductionType =
+  | "Option"
+  | "RepetitionMandatory"
+  | "RepetitionMandatoryWithSeparator"
+  | "Repetition"
+  | "RepetitionWithSeparator"
+
+export type LookaheadProductionType = OptionalProductionType | "Alternation"
+
+/**
+ * Computes all lookahead paths for a given production.
+ *
+ * The result is a three-dimensional array of token types.
+ * Accessing this array with the index of an alternative of the given production returns a two-dimensional array.
+ * Each entry of this array represents a list of the token types which can occur in this alternative.
+ *
+ * @param options.occurrence The occurrence number of this production within its rule.
+ *
+ * @param options.rule The rule which contains this production.
+ *
+ * @param options.prodType The type of this production.
+ *
+ * @param options.maxLookahead The maximum amount of lookahead for this production.
+ */
+export declare function getLookaheadPaths(options: {
```

Comment:
For one, the quirks test uses it, see https://github.com/Chevrotain/chevrotain/blob/d56c7705992d3f4c8fdd41bf2c307b8565f9850e/packages/chevrotain/test/full_flow/ecma_quirks/ecma_quirks.ts#L100-L105

In the same vein, any lookahead strategy would use the method in a similar manner. For example, the LL-Star lib would use the function to determine whether a production is LL(1) decidable, so that it doesn't have to go through its slower LL(*) algorithm.